Added iOS template for the example applications (#14)
heremaps/here-olp-edge-sdk-cpp: docs/dataservice-write-example.md

Note, that you can run `CMake` command directly from `<olp-edge-sdk-root>/examples/dataservice-write/` folder if you have already built and installed OLP C++ EDGE SDK libraries for Android. Make sure that you pass the correct path to `LevelDB` library and provided the correct `EDGE_SDK_NETWORK_PROTOCOL_JAR` parameter in the `CMake` command invoked by the `build/examples/dataservice-write/android/app/build.gradle` script.

## Building and running on iOS

> This example shows how to integrate and use the OLP C++ EDGE SDK in a basic iOS application written in Objective-C language.

Reviewer: I just notice that we use here `OLP C++ EDGE SDK` but in README.md `HERE OLP Edge SDK`
We need to align. Please change all occurrences to `HERE OLP Edge SDK`

Author: Done

> 

Reviewer: Please follow same structure as for Android.
Means:
### Prerequisities
### Build...
### Assemble... & Additional notes [IF NEEDED]

Author: Done

Configure the OLP C++ EDGE SDK with `EDGE_SDK_BUILD_EXAMPLES` variable set to `ON`, disable tests via `EDGE_SDK_ENABLE_TESTING` variable, and specify path to the iOS toolchain file shipped with SDK (located in `<olp-edge-sdk-root>/cmake/toolchains/iOS.cmake`):

> 

Reviewer: ```suggestion
First, before building, you need to configure the HERE OLP Edge SDK with `EDGE_SDK_BUILD_EXAMPLES` set to `ON`, disable tests with `EDGE_SDK_ENABLE_TESTING` set to `OFF`, and specify the path to the iOS toolchain file shipped together with the SDK and located under `${PROJECT_SOURCE_DIR}/cmake/toolchains/iOS.cmake`:
```

Author: Done

```bash
mkdir build && cd build
cmake .. -GXcode  -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/iOS.cmake -DPLATFORM=iphoneos -DEDGE_SDK_BUILD_EXAMPLES=ON -DEDGE_SDK_ENABLE_TESTING=OFF
```

> (note that in order to configure the OLP C++ EDGE SDK for a simulator, set `SIMULATOR` variable to `ON`)

Reviewer: Remove braces.

> (note that in order to configure the OLP C++ EDGE SDK for a simulator, set `SIMULATOR` variable to `ON`)

Reviewer: ```suggestion
Note that in order to configure the HERE OLP Edge SDK C++ for a simulator, you need to set the `SIMULATOR` variable to `ON`.
```

Author: Done

Now open the generated `XCode` project:
```bash
open olp-cpp-sdk.xcodeproj
```

> Select the `dataservice-write-example` scheme from the schemes list in `XCode` and specify correct your team identifier for the `dataservice-write-example` target.

Reviewer: ```suggestion
Select the `dataservice-write-example` scheme from the schemata list in `XCode` and specify your correct team identifier for the `dataservice-write-example` target.
```

Author: I changed a bit this sentence. Btw, what wrong with schemes list? Schemata is plural for schema, but XCode uses **scheme** notion:  https://developer.apple.com/library/archive/featuredarticles/XcodeConcepts/Concept-Schemes.html

Reviewer: I thought it was meant to be schema.
Ignore my comment then.

> Once everything is correctly setup, build and run the example application on your device and you should see the `Publish Successful` message in the main UI screen. If you encountered an error (`Publish Failed` message), please check the device's logs for the detailed description of the error message.

Reviewer: ```suggestion
Once everything is correctly set up, build and run the example application on your device and you should see the `Publish Successful` message in the main UI screen. If you encountered an error message, e.g. `Publish Failed`, please check the device's logs for the detailed description of the error.
```

Author: Done
